### PR TITLE
Add various development notes

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This aims to provide more detail for someone looking to develop the automation systems as well as the simulator in general.

### CODE REVIEW

**Comment by WillB97** [comment]
Do we want to add a note about how to create robot archives as well?

**Review by RealOrangeOne — APPROVED** [review]

**Comment by PeterJCLaw** [comment]
> Do we want to add a note about how to create robot archives as well?

Done in ee06104e6683404783dfcc57100e84265aa32e59. In hindsight I'm not sure that's actually that useful.

I think the next step here is probably to re-evaluate the README as a whole and re-structure it as more general documentation. There's a lot of things I've ended up covering here from a development standpoint which should actually just be plain documented. I think this is better than nothing for now though.

**Review by WillB97 — APPROVED** [review]

**Comment by WillB97** [comment]
We should probably mention that the ARENA_ROOT environment variable cannot be passed to `run-comp-match`

File: README.md [hunk 1]
```diff
@@ -43,6 +43,82 @@ pip install -r script/typing/requirements.txt
...
+└── ultrasounds  # Some robot code for testing ultrasound sensors
+    └── robot.py
+```
+
+Given a setup like the above, running Webots such that it picks up on one of the
+arena directories is possible through setting the `ARENA_ROOT` when launching
+the webots process:
+
+```
+$ ARENA_ROOT=robots/brakes-arena webots --mode=pause worlds/Arena.wbt
```
Comment (PeterJCLaw) [inline README.md:78]: Need to clarify here that the given path must be an absolute path.

Comment (PeterJCLaw) [reply README.md]: a78aec9

**Comment by PeterJCLaw** [comment]
> We should probably mention that the ARENA_ROOT environment variable cannot be passed to `run-comp-match`

done in a78aec9

**Review by WillB97 — APPROVED** [review]

**Pull request merged**